apps/remix-ide/src/app/plugins/code-format.ts

```diff
@@ -0,0 +1,77 @@
+'use strict'
+import { Plugin } from '@remixproject/engine'
+import prettier from 'prettier/standalone'
+import { Options } from 'prettier';
+import sol from './code-format/index'
+import * as ts from 'prettier/parser-typescript'
+import * as babel from 'prettier/parser-babel'
+import * as espree from 'prettier/parser-espree'
+import path from 'path'
+
+const profile = {
+    name: 'codeFormatter',
+    desciption: 'prettier plugin for Remix',
+    methods: ['format'],
+    events: [''],
+    version: '0.0.1'
+}
+
+export class CodeFormat extends Plugin {
+
+    constructor() {
+        super(profile)
+    }
+
+    async format() {
+        const file = await this.call('fileManager', 'getCurrentFile')
+        const content = await this.call('fileManager', 'readFile', file)
+        let parserName = ''
+        let options: Options = {
+        }
+        switch (path.extname(file)) {
```

Comment:
at best we should have configuration files (for sol, js, ts).
https://github.com/prettier-solidity/prettier-plugin-solidity#configuration-file
but that can certainly be done in an other iteration.